src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -778,6 +778,48 @@ public async Task GoAwayFrame_UnprocessedStreamFirstRequestWaitsUntilSecondFinis
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task GoAwayFrame_RequestInFlight_Finished()
+        {
+            using (await Watchdog.CreateAsync())
```

Comment:
Do we have reason believe this is going to fail?  Seems like this line should be deleted.  We only added it to tests that were failing in order to get more diagnostic data in CI, and we should be removing it from those once we get such data.

Reply:
(It'd be great if there were a good way to automatically add it to all async tests, but I've not spent the time investigating xunit extensibility features to understand if/how that's possible, and I don't want to add this line to every async test.)

Reply:
@stephentoub I think we could plug that through test discoverer similarly as we do with XML tests:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Xml/ModuleCore/XunitRunner.cs
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Xml/ModuleCore/XunitTestCase.cs

I'm not sure if it's possible to do so without the extra attribute (i.e. NetworkingFact or something) - also it might be tricky to add watchdog specifically because `await` has to happen in the test